### DISCUSSION TITLE
Simplify access to the Windows file system

### DISCUSSION THREAD

**Discussion by tonycody** [microsoft/WSL#8990] [General]
```shell
╭─root@XS-WanFengDesktop in /mnt/d/.repos/.github/xs-fork/dubbo on jdk19 ✔ (origin/jdk19)
╰$
```

As above: 
When I need a long path to access files in D drive, can the prefix `/ mnt/ d` be simplified to, such as` / d / `?

Or provide built-in commands to directly access Windows common directories. 
For example: 
C Driver: wdc. 
D Driver: wdd. 
E Driver:wde. 
User Profile Home:wup. 
User Home Download:wud

**Comment by Biswa96** [comment]
It can be changed with automount.root option in wsl.conf file. See https://learn.microsoft.com/en-us/windows/wsl/wsl-config#wslconf

**Comment by oldium** [comment]
Put the following in `/etc/wsl.conf`:

```
[automount]
root = /
```

There are more interesting options (like `metadata` automount option), please have a look at the documentation @Biswa96 linked.

**Comment by tonycody** [comment]
@oldium @Biswa96 
It is still not convenient enough
`
root = /
`
It only solves the simplification of drive access, but it is like the access of common directories in Windows.

I think some need to provide environment variables, so that there are relevant standards for the adaptation of multiple environments for the development of some projects. 
For example, compile the directory to get Windows UserProfile under WSL. 
These environment variables should also exist in WSL2 Linux.

**Comment by Biswa96** [comment]
> For example, compile the directory to get Windows UserProfile under WSL.

Set the WSLENV variable like this in Command Prompt `set WSLENV=USERPROFILE/p` and then run wsl.exe. The user profile environment variable will be passed to wsl. See https://devblogs.microsoft.com/commandline/share-environment-vars-between-wsl-and-windows/

**Comment by tonycody** [comment]
> > For example, compile the directory to get Windows UserProfile under WSL.
> 
> Set the WSLENV variable like this in Command Prompt `set WSLENV=USERPROFILE/p` and then run wsl.exe. The user profile environment variable will be passed to wsl. See https://devblogs.microsoft.com/commandline/share-environment-vars-between-wsl-and-windows/

Yes, why not bring these commonly used ones directly to WSL2? 
We still need to set it separately!

**Comment by oldium** [comment]
> @oldium @Biswa96 It is still not convenient enough `root = /` It only solves the simplification of drive access, but it is like the access of common directories in Windows.

If you need a better/faster access to common directories, create links from WSL to your Windows folders you need.

**Comment by Biswa96** [comment]
> Yes, why not bring these commonly used ones directly to WSL2?

Because someone may use the USERPROFILE variable in WSL2 with different value irrespective of what the value is in Windows side.
